File: src/main/java/org/goafabric/dbagentnew/persistence/entity/Allergy.java
package org.goafabric.dbagentnew.persistence.entity;

import jakarta.persistence.*;


@Entity
@Table(name="allergy")
public class Allergy {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Version
    private Long version;

    private String allergy;

    public Allergy(String id, Long version, String allergy) {
        this.id = id;
        this.version = version;
        this.allergy = allergy;
    }

    Allergy() {
    }

    public String getId() {
        return id;
    }

    public Long getVersion() {
        return version;
    }

    public String getAllergy() {
        return allergy;
    }

    @Override
    public String toString() {
        return "AllergyEo{" +
                "id='" + id + '\'' +
                ", version=" + version +
                ", allergy='" + allergy + '\'' +
                '}';
    }
}

File: src/main/kotlin/org/goafabric/llm/dbassistant/persistence/entity/Allergy.kt
package org.goafabric.llm.dbassistant.persistence.entity

import jakarta.persistence.*

@Entity
@Table(name = "allergy")
class Allergy {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null
        private set

    @Version
    var version: Long? = null
        private set

    var allergy: String? = null
        private set

    constructor(id: String?, version: Long?, allergy: String?) {
        this.id = id
        this.version = version
        this.allergy = allergy
    }

    internal constructor()

    override fun toString(): String {
        return "AllergyEo{" +
                "id='" + id + '\'' +
                ", version=" + version +
                ", allergy='" + allergy + '\'' +
                '}'
    }
}
